domain/application/watcher_test.go

```diff
@@ -907,3 +1055,9 @@ func (s *stubCharm) Revision() int {
 func (s *stubCharm) Version() string {
 	return ""
 }
+
+type stubRevoker struct{}
```

Comment:
Nit: This should be called `NoOpRevoker`. I understand that you copied from the others in this file.